skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,308 @@
+"""
+Stockholm format (:mod:`skbio.io.format.stockholm`)
+===================================================
+
+.. currentmodule:: skbio.io.format.stockholm
+
+The Stockholm format is a multiple sequence alignment format written in Markup
+as opposed to a simple text-based format. Data is stored on separate lines,
+each with a unique data 'name' preceding it. The alignment can just contain
+data, or contain data and related metadata.
+
+An example Stockholm file, taken from [1]_:
+
+.. code-block:: none
+
+    # STOCKHOLM 1.0
+    #=GF ID    UPSK
+    #=GF SE    Predicted; Infernal
+    #=GF SS    Published; PMID 9223489
+    #=GF RN    [1]
+    #=GF RM    9223489
+    #=GF RT    The role of the pseudoknot at the 3' end of turnip yellow mosaic
+    #=GF RT    virus RNA in minus-strand synthesis by the viral RNA-dependent \
+RNA
+    #=GF RT    polymerase.
+    #=GF RA    Deiman BA, Kortlever RM, Pleij CW;
+    #=GF RL    J Virol 1997;71:5990-5996.
+    AF035635.1/619-641             UGAGUUCUCGAUCUCUAAAAUCG
+    M24804.1/82-104                UGAGUUCUCUAUCUCUAAAAUCG
+    J04373.1/6212-6234             UAAGUUCUCGAUCUUUAAAAUCG
+    M24803.1/1-23                  UAAGUUCUCGAUCUCUAAAAUCG
+    #=GC SS_cons                   .AAA....<<<<aaa....>>>>
+    //
+
+Format Support
+==============
+**Has Sniffer: Yes**
+
+**State: Experimental as of 0.4.0-dev.**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |No    |:mod:`skbio.alignment.TabularMSA`                              |
++------+------+---------------------------------------------------------------+
+
+Format Specification
+====================
+The Stockholm format contains two types of data. The first can contain raw DNA,
+RNA, or Protein data and the second is comprised of associated metadata.
+Raw data lines begin with an associated 'name', which
+often times is an id comprised of letters and numbers, such as
+'M24804.1/82-104' ([1]_). After the id and a few tab characters,
+the data is displayed. Metadata lines, however, begin with a '#' and a
+two-letter marker which describes the metadata type, for example '#=GF' (see
+Metadata Types). Each metadata line also contains an extra two-letter tag,
```

Comment:
yeah, that's what I would do.

On Wed, Jan 13, 2016 at 11:53 AM, Kestrel Gorlick <notifications@github.com>
wrote:

> In skbio/io/format/stockholm.py
> <https://github.com/biocore/scikit-bio/pull/1243#discussion_r49641559>:
>
> > ++------+------+---------------------------------------------------------------+
> > +|Reader|Writer|                          Object Class                         |
> > ++======+======+===============================================================+
> > +|Yes   |No    |:mod:`skbio.alignment.TabularMSA`                              |
> > ++------+------+---------------------------------------------------------------+
> > +
> > +Format Specification
> > +====================
> > +The Stockholm format contains two types of data. The first can contain raw DNA,
> > +RNA, or Protein data and the second is comprised of associated metadata.
> > +Raw data lines begin with an associated 'name', which
> > +often times is an id comprised of letters and numbers, such as
> > +'M24804.1/82-104' ([1]_). After the id and a few tab characters,
> > +the data is displayed. Metadata lines, however, begin with a '#' and a
> > +two-letter marker which describes the metadata type, for example '#=GF' (see
> > +Metadata Types). Each metadata line also contains an extra two-letter tag,
>
> So you think it would be more clear to call them "features" instead of
> "tags"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1243/files#r49641559>.
>